examples/nlp/question-answering.py

```diff
@@ -0,0 +1,422 @@
+to use your own. Because the built-in loss is computed internally during the forward
+pass, when using it you may find that some Keras metrics misbehave or give unexpected
+outputs. This is an area of very active development in 🤗 Transformers, though, so
+hopefully we'll have a good solution to that issue soon!
+
+For now, though, let's use the built-in loss without any metrics. To get the built-in
+loss, simply leave out the `loss` argument to `compile`.
+"""
+
+# Optionally uncomment the next line for float16 training
+tf.keras.mixed_precision.set_global_policy("mixed_float16")
+
+model.compile(optimizer=optimizer)
+
+"""
+And now we can train our model. Note that we're not passing separate labels - the labels
+are keys in the input dict, to make them visible to the model during the forward pass so
+it can compute the built-in loss.
+"""
+
+model.fit(train_set, validation_data=validation_set, epochs=2)
+
+"""
+And we're done! Let's give it a try, using some text from the keras.io frontpage:
+"""
+
+context = """Keras is an API designed for human beings, not machines. Keras follows best
+practices for reducing cognitive load: it offers consistent & simple APIs, it minimizes
+the number of user actions required for common use cases, and it provides clear &
+actionable error messages. It also has extensive documentation and developer guides. """
+question = "What is Keras?"
+
+inputs = tokenizer([context], [question], return_tensors="np")
+outputs = model(inputs)
+start_position = tf.argmax(outputs.start_logits, axis=1)
+end_position = tf.argmax(outputs.end_logits, axis=1)
+print(int(start_position), int(end_position[0]))
+
+"""
+Looks like our model thinks the answer is the span from tokens 1 to 12 (inclusive). No
+prizes for guessing which tokens those are!
+"""
+
+answer = inputs["input_ids"][0, int(start_position) : int(end_position) + 1]
+print(answer)
+
+"""
+And now we can use the `tokenizer.decode` method to turn those token IDs back into text:
+"""
+
+print(tokenizer.decode(answer))
+
+"""
+And that's it! Remember that this example was designed to be quick to run rather than
+state-of-the-art, and the model trained here will certainly make mistakes. If you use a
+larger model to base your training on, and you take time to tune the hyperparameters
+appropriately, you'll find that you can achieve much better losses (and correspondingly
+more accurate answers).
+"""
+
+"""
+Finally, you can push the model to the HuggingFace Hub. By pushing this model you will
+have,
+- A nice model card generated for you containing hyperparameters and metrics of the model
+training,
+- A web API for inference calls,
+- A widget in the model page that enables others to test your model.
+This model is currently hosted [here](https://huggingface.co/keras-io/transformers-qa)
+and we have prepared a separate neat UI for you
+[here](https://huggingface.co/spaces/keras-io/keras-qa).
+"""
+
+model.push_to_hub("transformers-qa", organization="keras-io")
```

Comment:
It seems questionable to push the model every time someones runs the script -- this should be kept as a code block in the text rather than as executable code